runtime/src/bank/bank_hash_details.rs

```diff
@@ -0,0 +1,244 @@
+//! Container to capture information relevant to computing a bank hash
+
+use {
+    super::Bank,
+    base64::{prelude::BASE64_STANDARD, Engine},
+    log::*,
+    serde::{
+        de::{self, Deserialize, Deserializer},
+        ser::{Serialize, SerializeSeq, Serializer},
+    },
+    solana_sdk::{
+        account::{Account, AccountSharedData, ReadableAccount},
+        clock::{Epoch, Slot},
+        hash::Hash,
+        pubkey::Pubkey,
+    },
+    std::str::FromStr,
+};
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub(crate) struct BankHashDetails {
+    /// client version
+    pub version: String,
+    pub account_data_encoding: String,
+    pub slot: Slot,
+    pub bank_hash: String,
+    pub parent_bank_hash: String,
+    pub accounts_delta_hash: String,
+    pub signature_count: u64,
+    pub last_blockhash: String,
+    pub accounts: BankHashAccounts,
+}
+
+impl BankHashDetails {
+    pub fn new(
+        slot: Slot,
+        bank_hash: Hash,
+        parent_bank_hash: Hash,
+        accounts_delta_hash: Hash,
+        signature_count: u64,
+        last_blockhash: Hash,
+        accounts: BankHashAccounts,
+    ) -> Self {
+        Self {
+            version: solana_version::version!().to_string(),
+            account_data_encoding: "base64".to_string(),
+            slot,
+            bank_hash: bank_hash.to_string(),
+            parent_bank_hash: parent_bank_hash.to_string(),
+            accounts_delta_hash: accounts_delta_hash.to_string(),
+            signature_count,
+            last_blockhash: last_blockhash.to_string(),
+            accounts,
+        }
+    }
+}
+
+impl TryFrom<&Bank> for BankHashDetails {
+    type Error = String;
+
+    fn try_from(bank: &Bank) -> Result<Self, Self::Error> {
+        let slot = bank.slot();
+        if !bank.is_frozen() {
+            return Err(format!(
+                "Bank {slot} must be frozen in order to get bank hash details"
+            ));
+        }
+
+        // This bank is frozen; as a result, we know that the state has been
+        // hashed which means the delta hash is Some(). So, .unwrap() is safe
+        let accounts_delta_hash = bank
+            .rc
+            .accounts
+            .accounts_db
+            .get_accounts_delta_hash(slot)
+            .unwrap()
+            .0;
```

Comment:
```suggestion
        let (accounts_delta_hash, ..) = bank
            .rc
            .accounts
            .accounts_db
            .get_accounts_delta_hash(slot)
            .unwrap();
```

nit: enum indexing is evil